MonsterTrainModdingAPI/Managers/CustomTriggerManager.cs

```diff
@@ -3,65 +3,179 @@
+
+                IEnumerable<Type> AllTriggers = AppDomain.CurrentDomain
+                    .GetAssemblies()
+                    .SelectMany(x => x.GetTypes())
+                    .Where(x => typeof(IMTTrigger).IsAssignableFrom(x) && x != typeof(IMTTrigger) && !x.ContainsGenericParameters);
+                //Get all valid IMTCharacterTrigger Types
+                IEnumerable<Type> CharacterTriggers = AllTriggers.Where(x => typeof(IMTCharacterTrigger).IsAssignableFrom(x) && x != typeof(IMTCharacterTrigger));
+                IEnumerable<Type> CardTriggers = AllTriggers.Where(x => typeof(IMTCardTrigger).IsAssignableFrom(x) && x != typeof(IMTCardTrigger));
+                //Iterate Through
+                foreach (Type type in CharacterTriggers)
+                {
+                    IMTCharacterTrigger myTrigger = (IMTCharacterTrigger)Activator.CreateInstance(type);
+                    MonsterTrainModdingAPI.API.Log(BepInEx.Logging.LogLevel.Debug, $"Registering: {myTrigger.LocalizationKey}");
+                    CharTriggerToNameDict[GetCharacterTrigger(myTrigger.ID)] = myTrigger.LocalizationKey;
+                    TypeToCharacterTriggerDict[type] = GetCharacterTrigger(myTrigger.ID);
+                }
+                //Iterate Through
+                foreach (Type type in CardTriggers)
+                {
+                    IMTCardTrigger myTrigger = (IMTCardTrigger)Activator.CreateInstance(type);
+                    MonsterTrainModdingAPI.API.Log(BepInEx.Logging.LogLevel.Debug, $"Registering: {myTrigger.LocalizationKey}");
+                    CardTriggerToNameDict[GetCardTrigger(myTrigger.ID)] = myTrigger.LocalizationKey;
+                    TypeToCardTriggerDict[type] = GetCardTrigger(myTrigger.ID);
+                }
+            }
+        }
+        /// <summary>
+        /// Gets a New Character Trigger GUID
         /// </summary>
         /// <returns></returns>
-        public static int GetNewGUID()
+        public static int GetNewCharacterGUID()
         {
-            NumTriggers++;
-            return NumTriggers;
+            NumCharTriggers++;
+            return NumCharTriggers;
+        }
+        /// <summary>
+        /// Gets a New Card Trigger GUID
+        /// </summary>
+        /// <returns></returns>
+        public static int GetNewCardGUID()
+        {
+            NumCardTriggers++;
+            return NumCardTriggers;
         }
         /// <summary>
         /// Registers a Localization String
         /// </summary>
         /// <typeparam name="T">The Trigger Type used to register for</typeparam>
+        /// <param name="RegisterName">the string to register for localization</param>
         /// <returns></returns>
-        public static string RegisterLocalizationString<T>(string RegisterName = "") where T:IMTCharacterTrigger
+        public static string RegisterCharacterLocalizationString<T>(string RegisterName = "") where T : IMTCharacterTrigger
```

Comment:
Misleading name? This is registering a character *trigger* localization string, right?

Also, is it expected that the actual localization is registered with the localization manager separately? Or does it not need to be?